src/main/java/com/amazon/opendistroforelasticsearch/security/privileges/PrivilegesEvaluator.java

```diff
@@ -167,6 +174,21 @@ public boolean isInitialized() {
         return configModel !=null && configModel.getSecurityRoles() != null && dcm != null;
     }
 
+    private void setUserInfoInThreadContext(User user, Collection<String> mappedRoles) {
+        if (threadContext.getTransient(OPENDISTRO_SECURITY_USER_INFO_THREAD_CONTEXT) == null) {
+            final ImmutableList.Builder<String> builder = ImmutableList.builder();
+            builder.add(user.getName(),
+                    String.join(",", user.getRoles()),
+                    String.join(",", Stream.concat(user.getOpenDistroSecurityRoles().stream(), mappedRoles.stream())
```

Comment:
can mappedRoles be null?

Reply:
AFAIK, no. It is either set to `injectedRoles` if it is not `null` or output of `mapRoles(user, caller)` that may return empty set but never returns `null`.